### PR TITLE
Frequency locks Confessional intercoms, but an emag can bypass it

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Confessional intercoms are locked to the confessional frequency, 148.1

## Why It's Good For The Game

Confessional intercoms anonymize your voice regardless of which channel it's on, which gives people an easy way to become unknown on the common channel where typically specialized equipment is required to accomplish that. (Such as a voice changer.)

## Changelog

:cl: Melbert
balance: The chapel's confessional intercom is locked to its initial frequency, but an emag could bypass it.
/:cl:

### CODE REVIEW

**Comment by itseasytosee** [comment]
I understqand that this is unintended, but is it really an issue?
It kind of just seems like a creative use of game mechanics that has both counterplay (checking the chapel) and an in universe explanation.

**Comment by tralezab** [comment]
i agree with itseasy, giving these niche areas of the game some showtime is probably good

**Comment by Shadowflame909** [comment]
I didn't even know this was a thing! Should become a wiki added feature since it gives a reason to be in the chapel, and adds a motive for learning channel frequencies

**Comment by Shadowflame909** [comment]
That's fair... But what if it became an emaggable unlocked functionality. Thus adding to the nature of subverting items for unintended uses.

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -372,7 +372,7 @@
 	data["frequency"] = frequency
 	data["minFrequency"] = freerange ? MIN_FREE_FREQ : MIN_FREQ
 	data["maxFrequency"] = freerange ? MAX_FREE_FREQ : MAX_FREQ
-	data["freqlock"] = freqlock
+	data["freqlock"] = freqlock != RADIO_FREQENCY_UNLOCKED
```
Comment (ShizCalev) [inline code/game/objects/items/devices/radio/radio.dm:375]:
```suggestion
	data["freqlock"] = freqlock == RADIO_FREQENCY_LOCKED
```

Comment (ShizCalev) [reply code/game/objects/items/devices/radio/radio.dm:375]: just... reads weird otherwise imo

Comment (ShizCalev) [inline code/game/objects/items/devices/radio/radio.dm:375]: shouldn't this be `== RADIO_FREQENCY_LOCKED` ?

**Pull request merged**